src/test/java/htsjdk/tribble/gff/Gff3WriterTest.java

```diff
@@ -0,0 +1,155 @@
+    private final Path ncbi_woodpecker_small = Paths.get(DATA_DIR + "ref_ASM69900v1_top_level.small.gff3");
+    private final Path feature_extends_past_circular_region = Paths.get(DATA_DIR + "feature_extends_past_circular_region.gff3");
+    private final Path with_fasta = Paths.get(DATA_DIR + "fasta_test.gff3");
+    private final Path with_fasta_artemis = Paths.get(DATA_DIR + "fasta_test_artemis.gff3");
+    private final Path ordered_cofeature = Paths.get(DATA_DIR, "ordered_cofeatures.gff3");
+    private final Path child_before_parent = Paths.get(DATA_DIR, "child_before_parent.gff3");
+    private final Path url_encoding = Paths.get(DATA_DIR, "url_encoding.gff3");
+
+    @DataProvider(name = "roundTripDataProvider")
+    public Object[][] roundTripDataProvider() {
+        return new Object[][] {
+                {ensembl_human_small , 5}, {gencode_mouse_small, 7}, {ncbi_woodpecker_small, 7}, {feature_extends_past_circular_region, 1}, {with_fasta, 2}, {with_fasta_artemis, 2},
+                {ordered_cofeature, 3}, {child_before_parent, 1}, {url_encoding, 1}
+        };
+    }
+
+    @Test(dataProvider = "roundTripDataProvider")
+    public void testRoundTrip(final Path path, final int expectedCompression) {
+
+        final HashSet<String> comments1 = new HashSet<>();
+        final HashSet<SequenceRegion> regions1 = new HashSet<>();
+        final LinkedHashSet<Gff3Feature> features1 = readFromFile(path, comments1, regions1);
+
+            //write out to temp files (one gzipped, on not)
+        try {
+            final Path tempFile = Files.createTempFile("gff3Writer", ".gff3");
+            final Path tempFileGzip = Files.createTempFile("gff3Writer", ".gff3.gz");
+
+            try (final Gff3Writer writer = new Gff3Writer(tempFile); final Gff3Writer writerGzip = new Gff3Writer(tempFileGzip)) {
+                for (final String comment : comments1) {
+                    writer.addComment(comment);
+                    writerGzip.addComment(comment);
+                }
+
+                for (final SequenceRegion region : regions1) {
+                    writer.addDirective(Gff3Codec.Gff3Directive.SEQUENCE_REGION_DIRECTIVE, region);
+                    writerGzip.addDirective(Gff3Codec.Gff3Directive.SEQUENCE_REGION_DIRECTIVE, region);
+                }
+
+                for (final Gff3Feature feature : features1) {
+                    writer.addFeature(feature);
+                    writerGzip.addFeature(feature);
+                }
+            } catch (final IOException ex) {
+                throw new TribbleException("Error writing gff3 files", ex);
+            }
+
+            //read temp files back in
+
+
+            final long unzippedSize = tempFile.toFile().length();
+            final long zippedSize = tempFileGzip.toFile().length();
+
+            Assert.assertTrue(unzippedSize > zippedSize * expectedCompression);
```

Comment:
This is kind of weird to be testing the compression ratio here?  Why not just test that the file is zipped?

Reply:
agreed, that's much nicer.  done